notebooks/data-sources/TestGrid/metrics/metric_template.ipynb

```diff
@@ -369,6 +369,115 @@
     "    flat_list = [item for sublist in percent_label_by_grid_csv for item in sublist]\n",
```

Comment:
Yes, there is a lot of repetition in the function because the first one was hardcoded for `tests[i]==Overall`. We would want to eventually remove the old function, but all our other metrics are using this function. So for now, let's keep them both and maybe later if this case was affecting other KPIs, we can remove it. 

And I can try changing the condition to see if that works well :+1:

Reply:
<p>Great point Shrey. This whole label wise encoding function looks like it needs some refactoring I think. We should probably separate the duration collection from the rest of it too. </p><p> </p><p>But I also agree with Aakanksha, let's not let this hold up this PR.   </p>

Reply:
<p>Adding my 2cents - if we really want to remove the duplication, maybe we can consider adding an arg to the original func like this <code>def testgrid_labelwise_encoding(data, label, overall_only=True)</code>. And then move the "overall" logic inside an if clause and add the new "all tests" logic in the else clause.</p><p> </p><pre class="ql-syntax" spellcheck="false">if overall_only:
	if tests[i] == "Overall":
		# rest of the original calculation for overall duration
else:
	try:
		graphs[i][0].keys()
		# rest of the new logic for all tests duration calculation
</pre><p> </p><p>This way existing notebooks wont be affected, and at the same time we have 2 functions doing very similar things.</p>

Reply:
@chauhankaranraj also a good suggestion, and I think we should go ahead and implement that for now as its better than just having two functions that do nearly the same thing. That, said, I think its still worth revisiting this whole `labelwise encoding` function in a separate PR and see if we can make it a bit less complicated.   

@aakankshaduggal can we add an argument to `testgrid_labelwise_encoding()` to control whether we get all the data or just the data for "overall"?

Reply:
Thanks @chauhankaranraj and @MichaelClifford 
Sure, I will work on this and try to make one function that does everything :+1:

Reply:
:+1: